### PR TITLE
Quickstarts shouldn't part of the distribution zip yet

### PR DESCRIPTION
**Pull request by ge0ffrey**

### CODE REVIEW

**Comment by rsynek** [comment]
@ge0ffrey "Transfer failed for https://repo.maven.apache.org/maven2/com/zaxxer/SparseBitSet/1.2/SparseBitSet-1.2.jar: Connection reset"

I think now it's time to add `-Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false` to the maven execution.

File: optaplanner-distribution/src/main/assembly/assembly-optaplanner.xml [hunk 1]
```diff
@@ -64,31 +64,32 @@
       </excludes>
     </fileSet>
     <!-- Quickstart -->
-    <fileSet><!-- Note: going outside the module dir is bad, but it is not fetching generated files -->
-      <directory>../optaplanner-quickstarts</directory>
-      <outputDirectory>quickstarts</outputDirectory>
-      <excludes>
-        <exclude>pom.xml</exclude>
+<!--    <fileSet>&lt;!&ndash; Note: going outside the module dir is bad, but it is not fetching generated files &ndash;&gt;-->
```
Comment (rsynek) [inline optaplanner-distribution/src/main/assembly/assembly-optaplanner.xml:67]: Do we need to keep the commented-out assembly together will all the exclusions?

Comment (ge0ffrey) [reply optaplanner-distribution/src/main/assembly/assembly-optaplanner.xml:67]: We decided that the zip requires the binaries of the quickstarts, but not the sources. So no, you're right. I'll remove this.

**Comment by ge0ffrey** [comment]
> @ge0ffrey "Transfer failed for https://repo.maven.apache.org/maven2/com/zaxxer/SparseBitSet/1.2/SparseBitSet-1.2.jar: Connection reset"
> 
> I think now it's time to add `-Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false` to the maven execution.

Separate PR issue I'd argue. You want to take that one?

**Comment by ge0ffrey** [comment]
@rsynek github actions failed again:
`
 [ERROR] Failed to execute goal on project optaplanner-persistence-jaxb: Could not resolve dependencies for project org.optaplanner:optaplanner-persistence-jaxb:bundle:7.39.0-SNAPSHOT: Could not transfer artifact org.glassfish.jaxb:txw2:jar:2.3.2 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/org/glassfish/jaxb/txw2/2.3.2/txw2-2.3.2.jar: Connection reset -> [Help 1]`

**Pull request merged**